packages/@react-aria/listbox/src/useOption.ts

```diff
@@ -117,7 +117,8 @@ export function useOption<T>(props: AriaOptionProps, state: ListState<T>, ref: R
   }
 
   if (isVirtualized) {
-    optionProps['aria-posinset'] = state.collection.getItem(key).index + 1;
+    let index = state.collection.getItem(key).index;
+    optionProps['aria-posinset'] = Number.isNaN(index) ? undefined : (index as number) + 1;
```

Comment:
Nit:
Doesn't `Number.isNaN` only return true if `index` is strictly `NaN`? Shouldn't we check that index is not undef as well since `index` is a optional property on a `Node`?

Reply:
yep, you are correct, i'll update it